assimilation_code/modules/assimilation/assim_tools_mod.f90

```diff
@@ -698,10 +705,14 @@ subroutine filter_assim(ens_handle, obs_ens_handle, obs_seq, keys,           &
       ! Convert the prior and posterior for this observation to probit space
       call transform_to_probit(grp_size, obs_prior(grp_bot:grp_top), dist_for_obs, &
          temp_dist_params, probit_obs_prior(grp_bot:grp_top), .false., &
-         bounded_below, bounded_above, lower_bound, upper_bound)
+         bounded_below, bounded_above, lower_bound, upper_bound, ierr)
+      ! If probit transform fails for any group, this observation cannot be assimimlated
+      if(ierr /= 0) cycle SEQUENTIAL_OBS
```

Comment:
I think it could be helpful to add a CAUTION comment here. 
Because of the new error check, the interaction between the transformations and the groups can become messy. If it happens and the probit transform fails, then only part of the ensemble may be updated for a certain obs. Because of the `cycle SEQUENTIAL_OBS` the remaining groups, which could be transformed without issues, are disregarded.